core/src/main/java/com/linecorp/armeria/server/healthcheck/CpuHealthChecker.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.DoubleSupplier;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * A {@link HealthChecker} that reports as unhealthy when the current
+ * CPU usage or CPU load exceeds threshold. For example:
+ * <pre>{@code
+ * final CpuHealthChecker cpuHealthChecker = HealthChecker.of(10, 10);<br>
+ * final boolean healthy = cpuHealthChecker.isHealthy();
+ * }</pre>
+ */
+// Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java">ProcessorMetrics.java</a> in the micrometer core.
+final class CpuHealthChecker implements HealthChecker {
+
+    private static final List<String> OPERATING_SYSTEM_BEAN_CLASS_NAMES = ImmutableList.of(
+            "com.ibm.lang.management.OperatingSystemMXBean", // J9
+            "com.sun.management.OperatingSystemMXBean" // HotSpot
+    );
+
+    private final OperatingSystemMXBean operatingSystemBean;
+
+    private final Class<?> operatingSystemBeanClass;
+
+    @Nullable
+    @VisibleForTesting
+    final Method systemCpuUsage;
+
+    private final double targetCpuUsage;
+
+    @Nullable
+    @VisibleForTesting
+    final Method processCpuUsage;
+
+    private final double targetProcessCpuLoad;
+
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    private static final ConcurrentHashMap<String, MethodHandle> CACHE = new ConcurrentHashMap<>();
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param targetCpuUsage the target cpu usage
+     * @param targetProcessCpuLoad the target process cpu usage
+     */
+    private CpuHealthChecker(double targetCpuUsage, double targetProcessCpuLoad) {
```

Comment:
Should we rename these to `cpuUsageThreshold` and `processCpuLoadThreshold`? 'target' sounds like we should aim to reach the 'target'.